ParkingZone/Areas/Admin/Controllers/ParkingSlotController.cs

```diff
@@ -10,15 +11,41 @@ namespace Parking_Zone.MVC.Areas.Admin.Controllers;
 public class ParkingSlotController : Controller
 {
     private readonly IParkingSlotService _parkingSlotService;
-    public ParkingSlotController(IParkingSlotService parkingSlotService)
+    private readonly IParkingZoneService _parkingZoneService;
+    public ParkingSlotController(IParkingSlotService parkingSlotService, IParkingZoneService parkinZoneService)
     {
         _parkingSlotService = parkingSlotService;
+        _parkingZoneService = parkinZoneService;
     }
 
-    public IActionResult Index(long parkingZoneId)
+    public IActionResult Index(long zoneId)
     {
-        var parkingSlots = _parkingSlotService.GetSlotsByZoneId(parkingZoneId);
-        var listItemVMs = ListOfSlotsVM.MapToVM(parkingSlots);
+        var parkingSlots = _parkingSlotService.GetSlotsByZoneId(zoneId);
+        var listItemVMs = ListOfSlotsVM.MapToVM(parkingSlots).ToList();
+        var zone = _parkingZoneService.GetById(zoneId);
+        ViewData["parkingZoneId"] = zoneId;
+        ViewData["name"] = zone.Name;
         return View(listItemVMs);
     }
+
+    public IActionResult Create(long parkingZoneId)
+    {
+        CreateVM createVM = new()
+        {
+            ParkingZoneId = parkingZoneId,
+        };
+        return View(createVM);
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public IActionResult Create(CreateVM createVM)
+    {
```

Comment:
write a method that checks for slot number. if I enter existing number in database it should give model state error